Makefile

```diff
@@ -39,9 +39,10 @@ GOARCH := $(shell go env GOARCH)
 
 VERBOSE ?= false
 
-SOURCE_GIT_TAG ?=$(shell git describe --tags --exclude latest 2>/dev/null || echo "v0.0.0-unknown")
-SOURCE_GIT_TREE_STATE ?=$(shell ( ( [ ! -d ".git/" ] || git diff --quiet ) && echo 'clean' ) || echo 'dirty')
-SOURCE_GIT_COMMIT ?=$(shell git rev-parse --short "HEAD^{commit}" 2>/dev/null || echo "unknown")
+# Check for ENV vars first (from container), fallback to git commands (for host builds)
+SOURCE_GIT_TAG ?= $(or $(shell echo $$SOURCE_GIT_TAG 2>/dev/null), $(shell git describe --tags --exclude latest 2>/dev/null || echo "v0.0.0-unknown"))
+SOURCE_GIT_TREE_STATE ?= $(or $(shell echo $$SOURCE_GIT_TREE_STATE 2>/dev/null), $(shell ( ( [ ! -d ".git/" ] || git diff --quiet ) && echo 'clean' ) || echo 'dirty'))
```

Comment:
Is there a scenario where we'd want to allow the `SOURCE_GIT_TREE_STATE` to be passed in as parameter rather than determining the true state here?